src/main/java/sirius/biz/jobs/batch/file/ArchiveImportJob.java

```diff
@@ -0,0 +1,130 @@
+                backupInputFile(file.name(), fileHandle);
+                zipFile = new ZipFile(fileHandle.getFile());
+                importEntries();
+            }
+        } else {
+            throw Exceptions.createHandled().withNLSKey("FileImportJob.fileNotSupported").handle();
+        }
+    }
+
+    protected abstract void importEntries() throws Exception;
+
+    /**
+     * Fetches an entry from the archive and returns it's input stream.
+     * <p>
+     * Note, that previous opened input stream might get closed by performing this action.
+     *
+     * @param fileName   the name of the file to fetch
+     * @param isRequired flag if the file is required
+     * @return input stream for the requested file
+     * @throws Exception in case of an exception during fetching or if the file wasn't found but is required
+     */
+    @Nullable
+    protected InputStream fetchEntry(String fileName, boolean isRequired) throws Exception {
+        Enumeration<? extends ZipEntry> zipEntries = zipFile.entries();
+        while (zipEntries.hasMoreElements()) {
+            ZipEntry zipEntry = zipEntries.nextElement();
+
+            if (Strings.areEqual(fileName, zipEntry.getName())) {
+                return zipFile.getInputStream(zipEntry);
+            }
+        }
+
+        handleMissingFile(fileName, isRequired);
+        return null;
+    }
+
+    protected void handleMissingFile(String fileName, boolean isRequired) {
+        if (isRequired) {
+            throw Exceptions.createHandled()
+                            .withNLSKey("ArchiveImportJob.errorMsg.requiredFileMissing")
+                            .set("fileName", fileName)
+                            .handle();
+        } else {
+            process.log(ProcessLog.warn()
+                                  .withNLSKey("ArchiveImportJob.errorMsg.optionalFileMissing")
+                                  .withContext("fileName", fileName)
+                                  .withMessageType(fileName));
+        }
+    }
+
+    /**
+     * Checks if all given files are found in the archive.
+     *
+     * @param fileNamesToCheck all file names that should be inside the archive
+     * @return <tt>true</tt> if the archive contains all the given files, <tt>false</tt> otherwise
+     */
+    protected boolean containsEntries(String... fileNamesToCheck) {
+        ArrayList<? extends ZipEntry> zipEntries = Collections.list(zipFile.entries());
+
+        return Arrays.stream(fileNamesToCheck)
+                     .allMatch(fileName -> zipEntries.stream().anyMatch(entry -> entry.getName().equals(fileName)));
+    }
+
+    @Override
+    protected final boolean canHandleFileExtension(@Nullable String fileExtension) {
+        return "zip".equalsIgnoreCase(fileExtension);
+    }
+
+    @Override
+    protected void executeForStream(String filename, InputStream in) throws Exception {
+        // never called in overwritten execute method
```

Comment:
```suggestion
        throw new UnsupportedOperationException();
```
? Dann knallts auch falls es doch mal aufgerufen wird